ops/pebble.py

```diff
@@ -1966,3 +1952,162 @@ def send_signal(self, sig: typing.Union[int, str], services: typing.List[str]):
             'services': services,
         }
         self._request('POST', '/v1/signals', body=body)
+
+
+class MultipartLargeFileParser:
+    """A limited purpose multi-part parser backed by files for memory efficiency."""
+
+    def __init__(self, boundary: bytes):
+        self._response = None
+        self._files = {}
+
+        # RFC 2046 says that the boundary string needs to be preceded by a CRLF.
+        # Unfortunately, the request library's header parsing logic strips off one of
+        # these, so we'll prime the buffer with that missing sequence.
+        self._buffer = b'\r\n'
+
+        # Prepare the MIME multipart boundary line patterns.
+        # Note: RFC 2046 notes optional "linear whitespace" (e.g. [ \t]) after the boundary pattern
+        # and the optional "--" suffix.  Pebble doesn't use this, so to simplify implementation I
+        # am ignoring it.
+        self._boundary = "\r\n--{}\r\n".format(boundary).encode()
+        self._terminal_boundary = "\r\n--{}--\r\n".format(boundary).encode()
+
+        # State vars, as we may enter the feed() function multiple times.
+        self._headers = None
+        self._part_type = None
+
+    def __del__(self):
+        for file in self._files.values():
+            os.unlink(file.name)
+
+    def feed(self, data: bytes):
+        """Provide more data to the parser to process."""
+        self._buffer += data
+        # self._run_parser_fsm()
```

Comment:
Nit: this commented-out line should be removed.

Reply:
removed.